geode-apis-compatible-with-redis/src/integrationTest/java/org/apache/geode/redis/internal/executor/sortedset/AbstractZRemRangeByLexIntegrationTest.java

```diff
@@ -0,0 +1,329 @@
+    String memberName = "member";
+    jedis.zadd(KEY, SCORE, memberName);
+
+    // Range -infinity <= member name <= n
+    assertThat(jedis.zremrangeByLex(KEY, "-", "[n")).isEqualTo(1);
+  }
+
+  @Test
+  public void shouldReturnOne_givenOneMemberAndMemberNameInRangeInclusiveMToPlus() {
+    String memberName = "member";
+    jedis.zadd(KEY, SCORE, memberName);
+
+    // Range m <= member name <= +infinity
+    assertThat(jedis.zremrangeByLex(KEY, "[m", "+")).isEqualTo(1);
+  }
+
+  @Test
+  public void shouldReturnOne_givenOneMemberInRange_MinEqualToMemberNameAndMinInclusive() {
+    String memberName = "member";
+    jedis.zadd(KEY, SCORE, memberName);
+
+    // Range member <= member name <= n
+    assertThat(jedis.zremrangeByLex(KEY, "[" + memberName, "[n")).isEqualTo(1);
+  }
+
+  @Test
+  public void shouldReturnOne_givenMaxEqualToMemberNameAndMaxInclusive() {
+    String memberName = "member";
+    jedis.zadd(KEY, SCORE, memberName);
+
+    // Range a <= member name <= member
+    assertThat(jedis.zremrangeByLex(KEY, "[a", "[" + memberName)).isEqualTo(1);
+  }
+
+  @Test
+  public void shouldReturnOne_givenMinAndMaxEqualToMemberNameAndMinAndMaxInclusive() {
+    String memberName = "member";
+    jedis.zadd(KEY, SCORE, memberName);
+
+    assertThat(jedis.zremrangeByLex(KEY, "[" + memberName, "[" + memberName)).isEqualTo(1);
+  }
+
+  @Test
+  @Parameters({"[", "(", "", "-", "+"})
+  public void shouldReturnOne_givenMemberNameIsSpecialCharacterInRange(String memberName) {
+    jedis.zadd(KEY, SCORE, memberName);
+
+    assertThat(jedis.zremrangeByLex(KEY, "[" + memberName, "[" + memberName)).isEqualTo(1);
+  }
+
+  @Test
+  public void shouldReturnZero_givenMinEqualToMemberNameAndMinExclusive() {
+    String memberName = "member";
+    jedis.zadd(KEY, SCORE, memberName);
+
+    // Range member < member name <= n
+    assertThat(jedis.zremrangeByLex(KEY, "(" + memberName, "[n")).isEqualTo(0);
+  }
+
+  @Test
+  public void shouldReturnZero_givenMaxEqualToMemberNameAndMaxExclusive() {
+    String memberName = "member";
+    jedis.zadd(KEY, SCORE, memberName);
+
+    // Range a <= member name < member
+    assertThat(jedis.zremrangeByLex(KEY, "[a", "(" + memberName)).isEqualTo(0);
+  }
+
+  @Test
+  public void shouldReturnZero_givenMemberNotInRange() {
+    String memberName = "member";
+    jedis.zadd(KEY, SCORE, memberName);
+
+    // Range member name <= n <= o
+    assertThat(jedis.zremrangeByLex(KEY, "[n", "[o")).isEqualTo(0);
+  }
+
+  @Test
+  public void shouldReturnNumMembersRemoved_givenMultipleMembersInRange_withInclusiveMinAndMax() {
+    List<String> members = populateSortedSet();
+
+    int minLength = 3;
+    int maxLength = 6;
+    String min = StringUtils.repeat(BASE_MEMBER_NAME, minLength);
+    String max = StringUtils.repeat(BASE_MEMBER_NAME, maxLength);
+
+    List<String> sublist = new ArrayList<>(members.subList(minLength - 1, maxLength));
```

Comment:
Could `sublist` be renamed "membersToRemove" or something along those lines, to make it clearer what it is?